### PR TITLE
Added Google Analytics tracking

### PR DESCRIPTION
**Pull request by leonyeo**
Analytics is not enabled on the SPS account, hence, analytics is currently linked to my personal account. I can then access for the analytics data and reports with your accounts as well.

### CODE REVIEW

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -92,6 +92,8 @@ function initMap() {
         updateRestaurantList(data.results);
       })
     }))
+
+    gtag('event', 'search', {'search_term': autocomplete.getPlace().name});
```
Comment (lolfuljames) [inline src/main/webapp/script.js]: With auto-gps now enabled, you'll have to check whether the location was from User location or not. (if PAC has a place, PAC is used, else user GPS is used)

File: src/main/webapp/index.html [hunk 2]
```diff
@@ -32,13 +40,13 @@
       </div>
       <div id="infowindow-restaurant-content">
         <div id="info-restaurant-name">${name}</div>
-        <a target="_blank" id="info-restaurant-dir">
+        <a target="_blank" id="info-restaurant-dir" onclick="gtag('event', 'get_directions', {'method': 'Google Maps'})">
           <img src="https://gstatic.com/images/icons/material/system/2x/directions_gm_blue_20dp.png">
         </a>
-        <a id="whatsapp-share">
+      <a id="whatsapp-share" onclick="gtag('event', 'share', {'method': 'Whatsapp'})">
```
Comment (lolfuljames) [inline src/main/webapp/index.html]: This indentation should be like the previous line

**Review by lolfuljames — CHANGES_REQUESTED** [review]

File: src/main/webapp/script.js (hunk 1 above)
Comment (leonyeo) [reply src/main/webapp/script.js]: Added separate event fired when GPS updates

Comment (lolfuljames) [reply src/main/webapp/script.js]: Sorry i think I wasn't being clear enough.

This gtag event is being fired every time the submit button is being pressed, however there is this scenario whereby the user presses the submit button without choosing a place from PAC.

PAC.getPlace will be null, resulting in an error.

**Review by lolfuljames — CHANGES_REQUESTED** [review]

File: src/main/webapp/script.js (hunk 1 above)
Comment (leonyeo) [reply src/main/webapp/script.js]: Ah I see I get your concern now

```
if (!userMarker) {
        // User disabled location and didn't choose a place
        window.alert("Please enter a valid location!");
        return;
}
```
Lines 80-83

If I am not wrong, according to the above lines, if the user doesnt select a valid place, the above check will fire and return out of the function, hence, the tracking event will not be sent. I believe that this should already address that scenario?

**Review by lolfuljames — APPROVED** [review]
LGTM

**Pull request merged**